rllib/agents/bandit/bandit.py

```diff
@@ -1,49 +1,102 @@
 import logging
-from typing import Type
+from typing import Type, Union
 
 from ray.rllib.agents.bandit.bandit_tf_policy import BanditTFPolicy
 from ray.rllib.agents.bandit.bandit_torch_policy import BanditTorchPolicy
-from ray.rllib.agents.trainer import Trainer, with_common_config
+from ray.rllib.agents.trainer import Trainer
 from ray.rllib.policy.policy import Policy
 from ray.rllib.utils.annotations import override
 from ray.rllib.utils.typing import TrainerConfigDict
+from ray.rllib.agents.trainer_config import TrainerConfig
+from ray.rllib.utils.deprecation import Deprecated
 
 logger = logging.getLogger(__name__)
 
-# fmt: off
-# __sphinx_doc_begin__
-DEFAULT_CONFIG = with_common_config({
-    # No remote workers by default.
-    "num_workers": 0,
-    "framework": "torch",
 
-    # Do online learning one step at a time.
-    "rollout_fragment_length": 1,
-    "train_batch_size": 1,
+class BanditConfig(TrainerConfig):
+    """Defines a contextual bandit configuration class from which
+    a contexual bandit algorithm can be built. Note this config is shared
+    between BanditLinUCBTrainer and BanditLinTSTrainer. You likely
+    want to use the child classes BanditLinTSConfig or BanditLinUCBConfig
+    instead.
 
-    # Make sure, a `train()` call performs at least 100 env sampling timesteps, before
-    # reporting results. Not setting this (default is 0) would significantly slow down
-    # the Bandit Trainer.
-    "min_sample_timesteps_per_reporting": 100,
-})
-# __sphinx_doc_end__
-# fmt: on
+    Example:
+        >>> config = BanditConfig().rollouts(num_rollout_workers=4).
```

Comment:
hey @smorad thanks for this PR! Can we please get the resp. `BanditConfig` import here?

Reply:
It isn't python to have two separate ways to do the same thing. Example removed and users should now use subclasses instead (which each have correct imports now).